### PR TITLE
Add the Plugins menu item back to free trial sites

### LINKED ISSUE #1191: Add the Plugins menu item back to free trial sites
On Woo Express Free Trial sites, the Plugins menu item was removed in a low-cost attempt to prevent users from installing/modifying plugins (in addition to enforcing this limitation in the back end).

Having to access to the Plugins menu item means that:

- End users (merchants) are not able to understand that they can install plugins to add functionality to their WordPress site.
- Developers are led to believe that they cannot install plugins on Woo Express (or even WordPress.com in general).

The goal here is to design and develop a Plugins landing page that:

1. Communicates the benefits of installing Plugins on a WordPress site.
2. Prompts users to upgrade to a paid plan.

### PR DESCRIPTION
**Pull request by xristos3490**
### Specification

This PR adds a new "Plugins" landing page for WX trial sites at the `domain.com/wp-admin/admin.php?page=plugins-upgrade` URL.

It also introduces the following track events:
- `free_trial_upgrade_now`
- `free_trial_browse_plugins`
- `free_trial_discover_extensions`
- `free_trial_get_inspired`

**Note:** Every track includes the `source: plugins` metadata.

Closes #1191 

### Validation

- Visit a WX trial site. 
- Ensure you see the "Plugins" landing page in the admin menu (and the expected priority.)
- Ensure the landing page doesn't appear for other WX plans (Essential, Performance) and the WPCOM's ecommerce plan.

### CODE REVIEW

**Review by ilyasfoo — APPROVED** [review]
Tested well. LGTM! 

I've also updated changelog, removed some line breaks, and updated versions in 95beeee2b57f24974c0d3068fcd3541bc26b0a8b.

File: includes/free-trial/class-wc-calypso-bridge-free-trial-plugins-screen.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	/**
+	 * Initialize hooks.
+	 */
+	protected function init() {
+		add_action( 'admin_menu', array($this, 'add_menu_page'));
+		add_filter( 'admin_body_class', function( $classes ) {
+				$screen = get_current_screen();
+				if ( $screen && 'toplevel_page_plugins-upgrade' === $screen->id ) {
+					$classes .= 'woocommerce_page_wc-bridge-landing-page woocommerce_page_wc-plugins-landing-page';
```
Comment (beaulebens) [inline includes/free-trial/class-wc-calypso-bridge-free-trial-plugins-screen.php:53]: Should this have leading or trailing spaces, to make sure spacing is always maintained?

Comment (ilyasfoo) [reply includes/free-trial/class-wc-calypso-bridge-free-trial-plugins-screen.php]: Good catch, @beaulebens! I've added leading and trailing spaces along with another instance of this filter in 883dfafcaa989bad6a2fe53d4f9df98059f91b17

**Comment by ilyasfoo** [comment]
image

Minor feedback, I just noticed that WordPress footer is displayed in the new plugins screen, whereas it isn't in extensions screen. Perhaps nothing significant but I'd like to note it nonetheless.

**Pull request merged**